src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -2933,6 +2962,555 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_AVX512F_Fixup:
+        case NI_AVX512F_FixupScalar:
+        case NI_AVX512F_VL_Fixup:
+        {
+            assert(sig->numArgs == 4);
+
+            op4 = impPopStack().val;
+            op3 = impSIMDPopStack();
+            op2 = impSIMDPopStack();
+            op1 = impSIMDPopStack();
+
+            retNode = gtNewSimdHWIntrinsicNode(retType, op1, op2, op3, op4, intrinsic, simdBaseJitType, simdSize);
+
+            if (!retNode->isRMWHWIntrinsic(this))
+            {
+                if (!op1->IsVectorZero())
+                {
+                    GenTree* zero = gtNewZeroConNode(retType);
+
+                    if ((op1->gtFlags & GTF_SIDE_EFFECT) != 0)
+                    {
+                        op1 = gtNewOperNode(GT_COMMA, retType, op1, zero);
+                    }
+                    else
+                    {
+                        op1 = zero;
+                    }
+
+                    retNode->AsHWIntrinsic()->Op(1) = op1;
+                }
+            }
+            break;
+        }
+
+        case NI_AVX512F_TernaryLogic:
+        case NI_AVX512F_VL_TernaryLogic:
+        {
+            assert(sig->numArgs == 4);
+
+            op4 = impPopStack().val;
+
+            if (op4->IsIntegralConst())
```

Comment:
```suggestion
            if (op4->IsIntegralConst() && opts.OptimizationEnabled())
```

Reply:
We don't limit such import handling for other HWIntrinsics (there are no such checks in all the hwintrinsic importation logic). The normalization and correct importation of various nodes is considered part of the mainline handling for them and has been since they were first introduced with GT_SIMD.

Reply:
Ok; mainly what I was after was the ability to make this logic testable with e.g. Fuzzlyn.

Reply:
This should still be testable by validating the `TernaryLogic` against the corresponding sequence of `a op b op c`, no?

Particularly since the combining logic added in a future PR will be T1 only.

Reply:
Fuzzlyn/Antigen work by comparing unoptimized code against optimized code. They do not compare different variants of the same C# source that is expected to produce the same result. Also, they don't support intrinsics, so it would only be something we could do in the future.

> Particularly since the combining logic added in a future PR will be T1 only.

Yes, we would be able to fuzz the folding logic if it was T1 only, but the logic here is also very complex.

Reply:
I'd be fine with revisiting in the future if it helps Fuzzlyn when it eventually gets intrinsic support.